### PR TITLE
PhantomData<T> does not require bounds on T

### PR DESCRIPTION
**Pull request by dtolnay**

### CODE REVIEW

File: serde_codegen/src/bound.rs [hunk 1]
```diff
@@ -79,6 +79,13 @@ pub fn with_bound<F>(
     }
     impl visit::Visitor for FindTyParams {
         fn visit_path(&mut self, path: &ast::Path, _id: ast::NodeId) {
+            if let Some(seg) = path.segments.last() {
+                if seg.identifier.name.as_str() == "PhantomData" {
```
Comment (oli-obk) [inline serde_codegen/src/bound.rs]: While it's not great to implement a custom type called Phantom data, i think this should check the rest of the path, too.

Comment (dtolnay) [inline serde_codegen/src/bound.rs]: I don't think we always get the full path here... but I will try it.

Comment (oli-obk) [inline serde_codegen/src/bound.rs]: Oh..we're in the ast, i keep forgetting... Well, it's good enough :)

**Comment by oli-obk** [comment]
@homu r+

**Pull request merged**